javaslang/src/test/java/javaslang/collection/VectorPropertyTest.java

```diff
@@ -146,12 +148,117 @@ public void shouldDropRight() {
+                    history = history.append(Tuple.of(expected, actual));
+                }
+
+                if (random.nextInt(100) < 50) {
+                    final int value = random.nextInt();
+                    expected = expected.prepend(value);
+                    actual = assertAreEqual(actual, value, Vector::prepend, expected);
+                    history = history.append(Tuple.of(expected, actual));
+                }
+
+                if (random.nextInt(100) < 30) {
+                    final int n = random.nextInt(expected.size() + 1);
+                    expected = expected.drop(n);
+                    actual = assertAreEqual(actual, n, Vector::drop, expected);
+                    history = history.append(Tuple.of(expected, actual));
+                }
+
+                if (random.nextInt(100) < 30) {
+                    final int n = random.nextInt(expected.size() + 1);
+                    expected = expected.take(n);
+                    actual = assertAreEqual(actual, n, Vector::take, expected);
+                    history = history.append(Tuple.of(expected, actual));
+                }
+
+                if (!expected.isEmpty()) {
+                    assertThat(actual.head()).isEqualTo(expected.head());
+                    assertThat(actual.tail().toJavaList()).isEqualTo(expected.tail().toJavaList());
+                    history = history.append(Tuple.of(expected, actual));
+                }
+
+                if (!expected.isEmpty()) {
+                    final int index = random.nextInt(expected.size());
+                    assertThat(actual.get(index)).isEqualTo(expected.get(index));
+                    history = history.append(Tuple.of(expected, actual));
+                }
+
+                if (random.nextInt(100) < 50) {
+                    if (!expected.isEmpty()) {
+                        final int index = random.nextInt(expected.size());
+                        final int value = random.nextInt();
+                        expected = expected.update(index, value);
+                        actual = assertAreEqual(actual, null, (a, p) -> a.update(index, value), expected);
+                        history = history.append(Tuple.of(expected, actual));
+                    }
+                }
+
+                for (int k = 0; k < 10; k++) {
+                    if (!expected.isEmpty()) {
+                        final int from = random.nextInt(expected.size());
+                        final int to = random.nextInt(expected.size());
+                        expected = expected.slice(from, to);
+                        actual = assertAreEqual(actual, null, (a, p) -> a.slice(from, to), expected);
+                        history = history.append(Tuple.of(expected, actual));
+                    }
+                }
+
+                history.forEach(t -> assertAreEqual(t._1, t._2)); // test that the modifications are persistent
```

Comment:
I'm even testing that the previous versions are all intact - just to be sure I didn't accidentally modify an existing array;